ocfweb/api/lab.py

```diff
@@ -0,0 +1,40 @@
+from django.http import JsonResponse
+from ocflib.infra.hosts import hostname_from_domain
+from ocflib.lab.stats import get_connection
+from ocflib.lab.stats import list_desktops
+
+from ocfweb.caching import cache
+from ocfweb.caching import periodic
+
+
+@cache()
+def _list_public_desktops():
+    return list_desktops(public_only=True)
+
+
+@periodic(5)
+def _get_desktops_in_use():
+    """List which desktops are currently in use."""
+
+    # https://github.com/ocf/ocflib/blob/master/ocflib/lab/ocfstats.sql#L70
+    # we don't use users_in_lab_count_public because we're looking for
+    # desktops in use, and the view does COUNT(DISTINCT users)
+    with get_connection() as c:
+        c.execute(
+            'SELECT `host` FROM `session_duration_public` '
+            'WHERE `end` IS NULL;',
+        )
```

Comment:
I feel like making a new SQL view to show free desktops would make this easier, then you could just get that view directly and could also get the count of the free desktops from just the count of the elements in that view.

Reply:
Another way we could create the view is that:
column1: host (list of all desktops)
column2: if_login (1/True means logged in, 0/False means not)